Dockerfile.e2e

```diff
@@ -1,5 +1,5 @@
 ## Step 1: Build tests
-FROM golang:1.8.3-alpine3.6 as builder
+FROM golang:1.8.4-alpine3.6 as builder
```

Comment:
This will currently fail, because updated docker images are not yet live on Docker Hub

/cc @tianon

Reply:
Should be live now!

Reply:
Image looks to be up now:

```
1.8.4-alpine3.6: Pulling from library/golang
88286f41530e: Already exists
3f27971ac235: Pull complete
102255e55471: Pull complete
baa88d0dbe5f: Pull complete
38ab85760b4b: Pull complete
fa44aa219fdc: Pull complete
Digest: sha256:c21963f3a84d7ffb93a2fc60c8426db810abc372e95995d933c06170b897cba4
Status: Downloaded newer image for golang:1.8.4-alpine3.6
```